web/hooks/use-contract-supabase.ts

```diff
@@ -26,6 +31,38 @@ export const usePublicContracts = (contractIds: string[] | undefined) => {
   return contracts
 }
 
+export function useRealtimeContract(contractId: string | undefined) {
+  const [contract, setContract] = useState<Contract | undefined | null>(
+    undefined
+  )
+  useEffectCheckEquality(() => {
+    if (contractId) {
+      getContract(contractId)
+        .then((result) => setContract(result))
+        .catch((e) => console.log(e))
+    }
+  }, [contractId])
+  const changesLength = useRef(0)
+  const changedContract = useRealtimeChanges('UPDATE', 'contracts', {
+    k: 'id',
+    v: contractId ?? '',
+  })
+  console.log(changedContract)
+
+  useEffectCheckEquality(() => {
+    if (!changedContract) {
+      return
+    }
+    const length = changedContract.length
+    if (length > changesLength.current) {
+      setContract(changedContract[length - 1].new.data as Contract)
+      changesLength.current = length
+    }
+  }, [changedContract])
+
+  return contract
+}
```

Comment:
This function looks surprisingly complicated. Why isn't it something like
```typescript
const { rows } = useSubscription('contracts', { k: 'id', v: contractId ?? '_' })
return rows != null && rows.length > 0 ? rows[0].data as Contract : undefined
```
It also is fishy that this function is getting called with `undefined` and implies that there's something wrong in the React component design that is calling this hook. You should consider factoring the design so that the components calling this hook don't render when there is no contract.